### DISCUSSION TITLE
Infinite scroll?

### DISCUSSION THREAD

**Discussion by vladstudio** [unpoly/unpoly#337] [Q&A] [answered]
Hi! I might be missing something obvious, but how do I implement an infinite scroll with Unpoly? (that is, send a request when an element is revealed). Example from HTMX: 
https://htmx.org/examples/infinite-scroll/

**Comment by triskweline** [comment, marked as answer]
This isn't built in, but you can implement it yourself.

For example, start out with a list of items and a link to the next page:

```html
<ul class="list">
  <li>Foo</li>
  <li>Bar</li>
  <li>Baz</li>
</div>	

<a class="next-page" href="/page2">Next page</a>
```

Nothing magical about this yet. Clicking the link will replace the entire page.

We can now write a compiler that progressively enhance the link to (1) update the existing page and (2) automatically be followed when it is scrolled into the viewport:

```js
up.compiler('a.next-page', function(link) {
  let observer = new IntersectionObserver(function() {
    // Append the new list to the existing one. Also update the next page link.
    up.follow(link, { target: '.list:after, .next-page' })

    // No longer auto-follow the link in case the user scrolls while it is loading.
    observer.disconnect()
  })

  // Follow the link when it is scrolled into view
  observer.observe(link)

  // Stop observing when the link is removed
  return () => observer.disconnect()
})
```

I haven't tested this.

**Reply by vladstudio** [reply]
thank you for spending time to code that :-)
